frontend/src/metabase/query_builder/components/DataSelector.jsx

```diff
@@ -172,35 +170,59 @@ export default class DataSelector extends Component {
     }
 
     renderDatabaseSchemaPicker() {
-        const { selectedSchema } = this.state;
-
-        let sections = this.state.databases.map(database => {
-            return {
+        const { databases, selectedSchema } = this.state;
+
+        let sections = databases
+            .filter(database =>
+                // filter out the saved questions "db" so we can present it
+                // differently
+                !database.is_saved_questions
+            )
+            .map(database => ({
                 name: database.name,
                 items: database.schemas.length > 1 ? database.schemas : []
-            };
-        });
+            }));
+
+        // do the opposite of what we just did and get a reference to the saved question "db"
+        // there will only ever be one of these hence [0]
+        const savedQuestionSection = databases.filter(db => db.is_saved_questions)[0]
```

Comment:
You can use `.find` here instead. 

I would maybe even have simplified this logic such that I only check if the last item of `databases` has `is_saved_questions` as currently saved questions seem to always be the last item in array. But this is more a comment than a modification request :)